File: example/play-framework-2.4.6-java/test/software/betamax/BetamaxTest.java
package software.betamax;

import software.betamax.junit.Betamax;
import software.betamax.junit.RecorderRule;
import org.junit.Rule;
import org.junit.Test;
import play.libs.ws.WS;
import play.libs.ws.WSResponse;

import java.io.File;

import static org.junit.Assert.assertEquals;
import static play.test.Helpers.*;

/**
 * Created by sean on 2/7/16.
 */
public class BetamaxTest {

    File f = new File("test/resources/betamax/tapes");
    Configuration configuration = ProxyConfiguration.builder().tapeRoot(f).ignoreLocalhost(false).build();
    @Rule public RecorderRule recorder = new RecorderRule(configuration);

    @Test
    @Betamax(tape="my tape")
    public void testInServer() {
        int port = 3333;
        running(testServer(port), () -> {
            WSResponse wsResponse = WS.url("http://localhost:" + port).get().get(5000);
            assertEquals("PLAY", wsResponse.getHeader("X-Betamax"));
            assertEquals(903, wsResponse.getStatus());
            assertEquals("Hello from Betamax", wsResponse.getBody()); // altered body to confirm playback
        });
    }
}

File: example/play-framework-2.4.6-scala/test/software/betamax/BetamaxTest.scala
package software.betamax

import software.betamax.specs2.RecordedInteraction
import org.specs2.mutable.Specification
import play.api.libs.ws.WS
import play.api.test.{DefaultAwaitTimeout, WithServer}

import scala.concurrent.Await


/**
  * Created by sean on 2/15/16.
  */
class BetamaxTest extends Specification with DefaultAwaitTimeout {
  "A recorded interation" should {
    "replay /" in RecordedInteraction("index") {
      val port = 3333
      new WithServer(port = port) {
        val response = Await.result(WS.url(s"http://localhost:$port").get(), defaultAwaitTimeout.duration)
        response.body must beEqualTo("Hello from Betamax")
      }
    }
  }
}
